profiler/profile.go

```diff
@@ -267,25 +284,37 @@ func (p *profiler) runProfile(pt ProfileType) ([]*profile, error) {
 	tags := append(p.cfg.tags.Slice(), pt.Tag())
 	filename := t.Filename
 	// TODO(fg): Consider making Collect() return the filename.
-	if p.cfg.deltaProfiles && t.SupportsDelta {
+	if p.cfg.deltaProfiles && t.Delta != nil {
 		filename = "delta-" + filename
 	}
 	p.cfg.statsd.Timing("datadog.profiling.go.collect_time", end.Sub(start), tags, 1)
 	return []*profile{{name: filename, data: data}}, nil
 }
 
-// deltaProfile derives the delta profile between curData and the previous
-// profile. If extra profiles are provided, they will be merged into the final
-// profile after computing the delta profile.
-func (p *profiler) deltaProfile(name string, delta *pprofutils.Delta, curData []byte, extra ...*pprofile.Profile) (*profile, error) {
+type deltaProfiler struct {
+	delta *pprofutils.Delta
```

Comment:
NIT: I think I'd prefer the delta config to be an argument to the `deltaProfiler.Delta()` function. Otherwise it's kind of odd to have two copies of the same `*pprofutils.Delta` with different pointer semantic in two different places (here and on the profile type struct). The pointer here is meaningless (could be removed), and the pointer on the profile type has an overloaded meaning to indicate that delta profiling is not being used.

Again, I think it would all be a bit clearer if there weren't two copies of this type with the same data.

It's still fine to have a type that abstract the state of the delta profiler rather than directly using `*pprofile.Profile` to prepare for the upcoming change.

Reply:
Per our discussion, I've done the following things:

* Make the `Delta` field for `profileType` a slice of `pprofutils.ValueType`s and rename it to `DeltaValues`.
* Change `pprofutils.Delta` so that it only computes deltas for the values it was explicitly given. This way, a length 0 `pprofutils.ValueType` means "don't compute delta profiles" so we don't overload the meaning of the `Delta` field.
* Make the `delta` field of `deltaProfiler` a value rather than a pointer.